src/components/Card.js

```diff
@@ -1,67 +1,81 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick, handleDeleteClick) {
-    this._name = data.name;
-    this._link = data.link;
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
+    this._title = data.name;
+    this._image = data.link;
+    this._id = data._id;
+    this.isLiked = data.isLiked || false;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
     this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
 
-    // Fetch DOM elements and store them as class fields
     this._cardTemplate = document.querySelector(this._cardSelector);
     this._cardElement = this._getCardElement();
-    this._likeButton = this._cardElement.querySelector(".cards__like-button");
-    this._deleteButton = this._cardElement.querySelector(
-      ".cards__delete-button"
-    );
-
-    // Bind event listeners
     this._setEventListeners();
+    this._updateLikeState();
   }
 
-  // Event listeners setup
   _setEventListeners() {
-    // Like button event listener
-    this._likeButton.addEventListener("click", () => {
-      this._handleLikeButton();
-    });
-
-    // Delete button event listener
-    this._deleteButton.addEventListener("click", () => {
-      // open are you sure popup here
-      this._handleDeleteClick(this);
-    });
```

Comment:
**Could be improved**

You had a correct code here last time. You shouldn't have changed it